### PR TITLE
Add cluster vm override commands

### PR DESCRIPTION
**Pull request by dougm**
- Add cluster.override.change command to set VM DRS and HA overrides

- Add cluster.override.remove to remove VM overrides

- Add cluster.override.info to view current VM overrides

- Add cluster.rule.create '-depends' flag to support VM-to-VM dependency rules

### CODE REVIEW

File: govc/cluster/vm/change.go [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+	if vm == nil {
+		return flag.ErrHelp
+	}
+
+	cluster, err := cmd.Cluster()
+	if err != nil {
+		return err
+	}
+
```
Comment (anfernee) [inline govc/cluster/vm/change.go]: do we need to check cluster here as well?

File: govc/cluster/vm/remove.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+	if vm == nil {
+		return flag.ErrHelp
+	}
+
+	cluster, err := cmd.Cluster()
+	if err != nil {
+		return err
+	}
+
```
Comment (anfernee) [inline govc/cluster/vm/remove.go]: ditto.

File: govc/test/cluster.bats [hunk 3]
```diff
@@ -99,6 +99,65 @@ load test_helper
...
   assert_success
 
-  run govc cluster.rule.remove -cluster DC0_C0 -name pod1
+  run govc cluster.rule.remove -cluster DC0_C0 -name pod1 -depend
   assert_failure # rule does not exist
+
+  run govc cluster.rule.create -cluster DC0_C0 -name my_deps -depend
+  assert_failure # requires 2 groups
+
+  run govc cluster.rule.create -cluster DC0_C0 -name my_deps -depend my_app my_db
```
Comment (anfernee) [inline govc/test/cluster.bats]: this is `-depend` or `-depends`?, as well as the 3 cases above this.

File: govc/cluster/vm/change.go (hunk 1 above)
Comment (dougm) [reply govc/cluster/vm/change.go]: No, the cluster flag will never return `nil, nil`. VirtualMachineFlag is optional in some commands, and can have multiple values in other commands.  So it's left to the caller to decide what to do if none are specified.

File: govc/test/cluster.bats (hunk 3 above)
Comment (dougm) [reply govc/test/cluster.bats]: Good catch, it is `-depends`, those tests were expected to fail, but they were failing because of an invalid flag name.

**Comment by dougm** [comment]
Thinking about the command names, esp. after adding `cluster.vm.remove`..
Currently we have:

| UI label | govc command |
|------------|----------------------|
| VM/Host Groups | cluster.group.* |
| VM/Host Rules | cluster.rule.* |
| VM Overrides | cluster.vm.* |

Thinking `cluster.vm.*` should be renamed to `cluster.override.*` in this PR.  Thoughts?

**Comment by dougm** [comment]
Sticking with the new name, code is unchanged otherwise.

**Pull request merged**